### PR TITLE
Add vtkExtractVOI filter

### PR DESCRIPTION
**Pull request by banesullivan**
Adds an `extract_subset` filter to extract volumes of interest from image data/uniform grids while preserving the data type

### CODE REVIEW

**Comment by akaszynski** [comment]
Is it possible to an example and test for this?

**Comment by banesullivan** [comment]
I added a trivial test, but I can also add an example

File: pyvista/core/filters.py [hunk 1]
```diff
@@ -3566,3 +3566,24 @@ def gaussian_smooth(dataset, radius_factor=1.5, std_dev=2.,
...
+        To use this filter set the VOI ivar which are i-j-k min/max indices
+        that specify a rectangular region in the data. (Note that these are
+        0-offset.) You can also specify a sampling rate to subsample the
+        data.
+
+        Parameters
+        ----------
+        voi : tuple(int)
+            Length 6 tuple of integers specifying the volume of interest in
+            i-j-k min/max indices.
```
Comment (imsodin) [inline pyvista/core/filters.py]: I'd add an illustration of the ordering here, like in
https://github.com/pyvista/pyvista/blob/a572fd76cc5fac46a2f7e8c5ce977edd35ed1175/pyvista/core/filters.py#L130

File: pyvista/core/filters.py [hunk 2]
```diff
@@ -3566,3 +3566,24 @@ def gaussian_smooth(dataset, radius_factor=1.5, std_dev=2.,
...
         return _get_output(alg)
+
+
+    def extract_subset(dataset, voi):
+        """Select piece (e.g., volume of interest).
+
+        To use this filter set the VOI ivar which are i-j-k min/max indices
+        that specify a rectangular region in the data. (Note that these are
+        0-offset.) You can also specify a sampling rate to subsample the
+        data.
```
[the remark this answers is not in the document]
Comment (banesullivan) [reply pyvista/core/filters.py]: ah, my bad. I totally overlooked that.

File: pyvista/core/filters.py (hunk 1 above)
Comment (banesullivan) [reply pyvista/core/filters.py]: done

File: pyvista/core/filters.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (banesullivan) [reply pyvista/core/filters.py]: fixed

**Comment by banesullivan** [comment]
Just added a really cool example:

Large volume:

download


A subset of magma chamber under volcano

download

**Pull request merged**